app/models/authenticator/httpd.rb

```diff
@@ -25,9 +29,15 @@ def user_authorizable_without_authentication?
       true
     end
 
-    def _authenticate(_username, _password, request)
-      request.present? &&
-        request.headers['X-REMOTE-USER'].present?
+    def _authenticate(username, password, request)
+      if !user_data_collected?(request) && request.present? && oidc_configured?
```

Comment:
It feels like this could inverted to `return` early and not call the `user_data_collected?` method twice